core/src/main/java/io/paradoxical/cassieq/dataAccess/MessageRepositoryImpl.java

```diff
@@ -178,6 +183,47 @@ public void deleteAllMessages(final BucketPointer bucket) {
     }
 
     @Override
+    public void deleteAllMessages(MessagePointer from, MessagePointer to) {
+
+        final int lastBucketNumber = to.toBucketPointer(queueDefinition.getBucketSize()).get().intValue();
+        final int firstBucketNumber = from.toBucketPointer(queueDefinition.getBucketSize()).get().intValue();
+
+        List<Integer> batchBucketsToDelete = new ArrayList<>();
+
+        final Iterator<Integer> bucketRangeIterator = IntStream.range(firstBucketNumber, lastBucketNumber + 1)
+                                                               .boxed()
+                                                               .iterator();
+
+        while (bucketRangeIterator.hasNext()) {
+            batchBucketsToDelete.add(bucketRangeIterator.next());
+
+            if (batchBucketsToDelete.size() == getDeleteBatchSize()) {
+                deleteAllMessagesInBuckets(batchBucketsToDelete);
```

Comment:
consider using `ImmutableList.copyOf(batchBucketsToDelete)` as the argument incase something implementation wise ever changes (async, etc.)